### ISSUE TITLE
Inclusion of MoMoLib in trunk

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1105]
**Reported by dietmarw on 15 May 2013 14:22 UTC**
In ba2f041349fe1333d2124bef5ccdcf68a033faa3 the MoMoLib was included in trunk. This ticket should discuss possible issues that need to be resolved before the general release.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1105

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 15 May 2013 14:45 UTC**
Here is a list to issues I think need fixing:
 1.  New types used by MoMoLib were simply introduced in e2676d3ae248c1811a90ba0ffbc41d3d4b4f17f9 to `Modelica.SIunits` without any previous discussion or respective ticket (there are already other types on the waiting list that were not added for a reason). Since we are still in the middle of deciding what makes sense to add to `Modelica.SIunits` the new types should really be moved to `Modelica.Media.Common` where all other media specific types reside.
 2.  The commited version contains documentation that points to its development status. This should  be cleaned up ASAP.
 3. The commited version contains commented code sections. This is something that should be avoided in a final release of the MSL.
 4. The documentation of the new included components does not follow the [MSL guide lines on Modelica code](https://build.openmodelica.org/Documentation/Modelica.UsersGuide.Conventions.ModelicaCode.Format.html)
 5. Records don't extend from `Modelica.Icons.Record`

The items named above shows why we normally should **not**  throw a new library into the trunk but evaluate and improve them externally for at least 6 month. I'm really not sure why this was not done in this case :-(.

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 16 May 2013 05:29 UTC**
Hi,

As far as I understand it the new models are not conform to the 
specification because they look up types in partial packages in 
simulation models:

```
5.3.2 Composite Name Lookup
...
The class we look inside may not be partial in a simulation model.
```
In the simulation model Modelica.Media.Examples.R134a.R134a1 you have
```
redeclare package Medium = Modelica.Media.R134a.R134a_ph
```
package R134a_ph contains:
```
redeclare function extends specificEntropy 
  protected
  Common.HelmholtzDerivs f "helmholtz derivatives";
  Common.PhaseBoundaryProperties liq "properties on liquid phase boundary";
  ...
end specificEntropy;
```
Common.PhaseBoundaryProperties contains these:
```
record PhaseBoundaryProperties
  Modelica.Media.Interfaces.PartialMedium.IsobaricExpansionCoefficient beta "isobaric expansion coefficient";
  Modelica.Media.Interfaces.PartialMedium.IsentropicExponent gamma "isentropic exponent";
end PhaseBoundaryProperties;
```
which have types accessed from a partial package.

So are these models conform to the specification or not?

Cheers,
Adrian Pop/

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 16 May 2013 07:02 UTC**
Replying to [comment:1 dietmarw]:
> Here is a list to issues I think need fixing:
>  1.  New types used by MoMoLib were simply introduced in e2676d3ae248c1811a90ba0ffbc41d3d4b4f17f9 to `Modelica.SIunits` without any previous discussion or respective ticket (there are already other types on the waiting list that were not added for a reason). Since we are still in the middle of deciding what makes sense to add to `Modelica.SIunits` the new types should really be moved to `Modelica.Media.Common` where all other media specific types reside.

Library officers decide what is included in their libraries and for this no discussion is needed in MAP-LIB (otherwise this would be not practical if every minor detail is discussed in the ticket system). I am library officer of SIunits.

I included three new types because they are used in new components included in Modelica.Media. The new types are included exactly that the place in SIunits where other similar types are present:

MolarDensity is added at the place where MolarMass, MolarVolume, MolarEnergy is present and therefore this is the right place.

DerPressureByDensity and DerPressureByTemperature are added at the place where several DerxxxByyyy are present.

There is a difference whether a new type shall be included that is nowhere used in MSL (I don't do this without proper reasoning or discussion), or whether a type is introduced that is used in MSL (must be defined somewhere).

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 16 May 2013 07:28 UTC**
Replying to [comment:1 dietmarw]:


> 3. The commited version contains commented code sections. This is something that should be avoided in a final release of the MSL.

This is your opinion. A developer may have another opinion, and it depends on the actual situation whether it is more practical to keep an out-commented code part for some time.


> 4. The documentation of the new included components does not follow the [MSL guide lines on Modelica code](https://build.openmodelica.org/Documentation/Modelica.UsersGuide.Conventions.ModelicaCode.Format.html)

I guess that no Modelica class in MSL follows 100 % the guidelines. Note, these are guidelines and not requirements. It is impossible to check by a human whether all parts of the guidelines are fulfilled. Therefore, a comment like "does not follow the guidelines" is quite useless. Please, point out the major issues that should be improved in the documentation

> 5. Records don't extend from `Modelica.Icons.Record`

I searched, but did not find this issue. E.g. the records introduced in Media.Commons extend from Modelica.Icons.Record. Again, such an unspecific bug report is quite useless. Please, be specific and point to the issue.

> 
> The items named above shows why we normally should **not**  throw a new library into the trunk but evaluate and improve them externally for at least 6 month. I'm really not sure why this was not done in this case :-(.

New components have been included in an existing library (Media). The library officer is responsible for the content that is introduced in his library. I have coordinated this with Hubertus (the library officer) and XRG (the developers). Hubertus agreed to put it in and Hubertus and XRG discussed how to include it and how to rename the classes. I was just implementing the result of this discussion because neither XRG nor Hubertus had currently time to work on this.

The MoMoLibcomponents have been added to the development branch in Jan 2013, and since then several improvements have been made on the development version. Finally, everything seemed to be o.k. and Pedantic Check with Dymola and Check with Simulation (with exception of one model) was successful. Furthermore, in the CleanSky project the functionality of the library was carefully evaluated. It then seemed ready to include the library in trunk. I don't understand why you use the term "throw a library in the trunk" in such a situation.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 16 May 2013 07:36 UTC**
Replying to [comment:2 adrpo]:

> Hi,
> 
> As far as I understand it the new models are not conform to the 
> specification because they look up types in partial packages in 
> simulation models:
> 
> ```
> 5.3.2 Composite Name Lookup
> ...
> The class we look inside may not be partial in a simulation model.
> ```
> In the simulation model Modelica.Media.Examples.R134a.R134a1 you have
> ```
> redeclare package Medium = Modelica.Media.R134a.R134a_ph
> ```
> package R134a_ph contains:
> ```
> redeclare function extends specificEntropy 
> protected
> Common.[wiki:HelmholtzDerivs] f "helmholtz derivatives";
> Common.[wiki:PhaseBoundaryProperties] liq "properties on liquid phase boundary";
> ...
> end specificEntropy;
> ```
> Common.[wiki:PhaseBoundaryProperties] contains these:
> ```
> record PhaseBoundaryProperties
> Modelica.Media.Interfaces.[wiki:PartialMedium].[wiki:IsobaricExpansionCoefficient] beta "isobaric expansion coefficient";
> Modelica.Media.Interfaces.[wiki:PartialMedium].[wiki:IsentropicExponent] gamma "isentropic exponent";
> end PhaseBoundaryProperties;
> ```
> which have types accessed from a partial package.
> 

The definition of Common.PhaseBoundaryProperties in the trunk is:

```
record PhaseBoundaryProperties 
  "thermodynamic base properties on the phase boundary"
  extends Modelica.Icons.Record;
  SI.Density d "density";
  SI.SpecificEnthalpy h "specific enthalpy";
  SI.SpecificEnergy u "inner energy";
  SI.SpecificEntropy s "specific entropy";
  SI.SpecificHeatCapacity cp "heat capacity at constant pressure";
  SI.SpecificHeatCapacity cv "heat capacity at constant volume";
  DerPressureByTemperature pt "derivative of pressure w.r.t. temperature";
  DerPressureByDensity pd "derivative of pressure w.r.t. density";
end PhaseBoundaryProperties;
```

You seemed to point to the wrong code? Please, can you clarify.

> 
> So are these models conform to the specification or not?
> 
> Cheers,
> Adrian Pop/

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 16 May 2013 07:54 UTC**
You are probably looking in the wrong file. It is in [source:Modelica/trunk/Modelica/Media/R134a.mo@6354 R134a.mo].

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 16 May 2013 07:59 UTC**

I think you're mistaken. This is the code in trunk Modelica/Media/R134a.mo, ba2f041349fe1333d2124bef5ccdcf68a033faa3.
I just double checked right now.
```
    record PhaseBoundaryProperties
      "thermodynamic base properties on the phase boundary"
      extends Modelica.Icons.Record;

      Modelica.SIunits.Density d "density";
      Modelica.SIunits.SpecificEnthalpy h "enthalpy";
      Modelica.SIunits.SpecificEnergy u "inner energy";
      Modelica.SIunits.SpecificEntropy s "entropy";
      Modelica.SIunits.SpecificHeatCapacity cp
        "heat capacity at constant pressure";
      Modelica.SIunits.SpecificHeatCapacity cv
        "heat capacity at constant volume";
      Modelica.SIunits.IsothermalCompressibility kappa "isentropic exponent";
      Modelica.SIunits.Velocity a "velocity of sound";
      Modelica.Media.Interfaces.PartialMedium.IsobaricExpansionCoefficient beta
        "isobaric expansion coefficient";
      Modelica.Media.Interfaces.PartialMedium.IsentropicExponent gamma
        "isentropic exponent";
      Real pt "derivative of pressure wrt temperature";
      Real pd "derivative of pressure wrt density";

    end PhaseBoundaryProperties;
```

Cheers,
Adrian Pop/

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 16 May 2013 08:02 UTC**
Replying to [comment:4 otter]:
> Replying to [comment:1 dietmarw]:
> 
> > 3. The commited version contains commented code sections. This is something that should be avoided in a final release of the MSL.
> 
> This is your opinion. A developer may have another opinion, and it depends on the actual situation whether it is more practical to keep an out-commented code part for some time.

We are talking about a new component not a changed one where it might still be valuable to keep old code-snippets. The release version of MSL should not serve as a container for out-commented development ideas this is what branches can be used for. 

> > 4. The documentation of the new included components does not follow the [MSL guide lines on Modelica code](https://build.openmodelica.org/Documentation/Modelica.UsersGuide.Conventions.ModelicaCode.Format.html)
> 
> I guess that no Modelica class in MSL follows 100 % the guidelines. Note, these are guidelines and not requirements. It is impossible to check by a human whether all parts of the guidelines are fulfilled. Therefore, a comment like "does not follow the guidelines" is quite useless. Please, point out the major issues that should be improved in the documentation

Since some time we are struggling to improve the quality of the MSL. True the current MSL does not follow the guidelines 100%, that's why we said that all that is **added** to it follows this so that we only need to worry about updating existing code. It's much easier to fix new things at once than to go over it after some time. You say it is impossible by a human to follow the guidelines? They are not that much to read really and also quite easy to follow. If we can not manage even that tiny bit then I really don't want to think about what else we don't follow when adding new models.

In particular I meant that:
 * Description strings should start with a capital letter.
 * HTML errors in the documentation (Search for `Modelica.Media` in the [tidy.filtered](https://build.openmodelica.org/Documentation/tidy.filtered) file that is generated by OpenModelica on a daily basis

> > 5. Records don't extend from `Modelica.Icons.Record`
> 
> I searched, but did not find this issue. E.g. the records introduced in Media.Commons extend from Modelica.Icons.Record. Again, such an unspecific bug report is quite useless. Please, be specific and point to the issue.

Yes sorry, this was actually Dymola not displaying the icons correctly in the browser view.


> > The items named above shows why we normally should **not**  throw a new library into the trunk but evaluate and improve them externally for at least 6 month. I'm really not sure why this was not done in this case :-(.
> 
> New components have been included in an existing library (Media). The library officer is responsible for the content that is introduced in his library. I have coordinated this with Hubertus (the library officer) and XRG (the developers). Hubertus agreed to put it in and Hubertus and XRG discussed how to include it and how to rename the classes. I was just implementing the result of this discussion because neither XRG nor Hubertus had currently time to work on this.
> 
> The MoMoLibcomponents have been added to the development branch in Jan 2013, and since then several improvements have been made on the development version. Finally, everything seemed to be o.k. and Pedantic Check with Dymola and Check with Simulation (with exception of one model) was successful. Furthermore, in the CleanSky project the functionality of the library was carefully evaluated. It then seemed ready to include the library in trunk. I don't understand why you use the term "throw a library in the trunk" in such a situation.

Well this ticket is the first one filed on the inclusion of MoMoLib. This should have happened back then when it actually came up for discussion. Only until some days ago it was not clear that it should be included as of now. So people/tool vendors didn't even have a chance to evaluate it properly. So even though it was available on some development branch does not automatically mean is going to be included right? It really has to do with having a proper transparent procedure in place so that people can give feedback on possible things that needs fixing BEFORE it goes into the trunk. This was unfortunately not the case. I'm rather careful with trusting statements like "in the CleanSky project the functionality of the library was carefully evaluated" as long as the details of the evaluation are not available. E.g., was it done with different tools or just one and are the tools following the Modelica Specification. Since MSL 3.2.1 must be 100% standard this is important. And checking is of much value. Also I could not see any test models being committed for the new Media types. 

I'm not against inclusion in general just a certain QA should be followed or we end up with similar problems like after the premature inclusion of Spice3.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 16 May 2013 08:24 UTC**
Replying to [comment:2 adrpo]:


> Hi,
> 
> As far as I understand it the new models are not conform to the 
> specification because they look up types in partial packages in 
> simulation models:
> 
> ```
> 5.3.2 Composite Name Lookup
> ...
> The class we look inside may not be partial in a simulation model.
> ```
> In the simulation model Modelica.Media.Examples.R134a.R134a1 you have
> ```
> redeclare package Medium = Modelica.Media.R134a.R134a_ph
> ```
> package R134a_ph contains:
> ```
> redeclare function extends specificEntropy 
> protected
> Common.[wiki:HelmholtzDerivs] f "helmholtz derivatives";
> Common.[wiki:PhaseBoundaryProperties] liq "properties on liquid phase boundary";
> ...
> end specificEntropy;
> ```
> Common.[wiki:PhaseBoundaryProperties] contains these:
> ```
> record PhaseBoundaryProperties
> Modelica.Media.Interfaces.[wiki:PartialMedium].[wiki:IsobaricExpansionCoefficient] beta "isobaric expansion coefficient";
> Modelica.Media.Interfaces.[wiki:PartialMedium].[wiki:IsentropicExponent] gamma "isentropic exponent";
> end PhaseBoundaryProperties;
> ```
> which have types accessed from a partial package.
> 
> So are these models conform to the specification or not?
> 
> 

Fixed in 88fe4779ec24d73ddff6f78f194092de2fe8559f.

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 16 May 2013 12:35 UTC**
Hi,

Thanks Martin O. 
Is not really clear what the specification allows in this case.
There are also other uses similar to this one in Modelica.Media i.e. #1106 which should be clarified.

Cheers,
Adrian Pop/

**Comment by modelica-trac-importer** [comment]
**Comment by leo&#46;gall on 16 May 2013 12:39 UTC**
Replying to [comment:8 dietmarw]:
> In particular I meant that:
>  \* Description strings should start with a capital letter.

I attached a report created by Dymola ModelManagement to this ticket. So it's clear, which description strings should be adapted.

I volunteer to make the trivial fixes, by changing lower case to capital letters (probably next week).

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#46;se on 16 May 2013 12:45 UTC**
Well. The main issue of MoMoLib I think should be resolved is:
MSL 3.2.1 is focusing on fixing old problems in MSL, making it conforming to the specification. It has taken a long time to get MSL reasonably stable and I am not sure adding a new library so short to the release date is a good idea.

It would be better to make a stable release, then add the proposed libraries and fix them for the next release. I am fairly certain all quirks will not be sorted out before the release (Adrian founds bugs within a few minutes of looking at the library despite it being checked with Dymola Pendantic).

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 16 May 2013 12:46 UTC**
Hi again,

I have another issue, this time with model:
```
Modelica.Media.Examples.RealGasAir.MoistAir1
```

You have in `Modelica.Media.Examples.RealGasAir.MoistAir1`:
```
redeclare package Medium = Modelica.Media.Air.RealGasMoistAir;
```
Then Modelica.Media.Air.RealGasMoistAir contains:
```
package RealGasMoistAir
  package Utilities "Utilitiy package for moist air"
    ...
    final constant MoleFraction[4] MMX={18.015257E-003,28.01348E-003, 31.9988E-003,39.948E-003};
 ...
  end Utilities;
end RealGasMoistAir;
```

However `MoleFraction` is not defined anywhere in any of the scopes accessible at this point even via import. Can you tell me how this really works in some other tools?
It should either be: `SI.MoleFraction` or `Types.MoleFraction`.

Cheers,
Adrian Pop/

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 16 May 2013 12:53 UTC**
Replying to [comment:11 leo.gall]:


> Replying to [comment:8 dietmarw]:
> 
> > In particular I meant that:
> > \* Description strings should start with a capital letter.
> > 
> 
> I attached a report created by Dymola ModelManagement to this ticket. So it's clear, which description strings should be adapted.
> 
> I volunteer to make the trivial fixes, by changing lower case to capital letters (probably next week).

Yes, please do it. I recognized that in the whole Modelica.Media library there are many descriptions strings that start with a lower case letters. Maybe, you can correct them as well.

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 16 May 2013 12:53 UTC**
Sorry. You can forget my last comment. `PartialMedium` extends from `Modelica.Media.Interfaces.Types` which contains `MoleFraction` so it should work fine.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 16 May 2013 13:05 UTC**
Replying to [comment:12 sjoelund.se]:


> Well. The main issue of MoMoLib I think should be resolved is:
> MSL 3.2.1 is focusing on fixing old problems in MSL, making it conforming to the specification. It has taken a long time to get MSL reasonably stable and I am not sure adding a new library so short to the release date is a good idea.

This is a misunderstanding. No new library is added. Instead three new medium models are added to the library Modelica.Media. These are just media models as all the others. The big issue for MSL 3.2.1 was that Modelica.Media was not fully compliant to the Modelica specification and it took some effort to make this compliant to each other and still backwards compatible. The newly introduced media have been adapted to this slightly changed library structuring, and there should be no problem. There are some compliance issues that are not reported by Dymola pedantic mode. Just run your tools to detect these and this can be quickly fixed.

> 
> It would be better to make a stable release, then add the proposed libraries and fix them for the next release. I am fairly certain all quirks will not be sorted out before the release (Adrian founds bugs within a few minutes of looking at the library despite it being checked with Dymola Pendantic).

How do you know that there are no problems with all the other media in Modelica.Media? This is a huge library and the only way to find this out is that a tool analysis the library and reports problems, and this can be performed within a few minutes. So, just run the same (automatic) tests again on Modelica.Media as run before and thats it (I run the once where I have easily access to it).

Note, there are still about 60 new components in the trunk (probably more). We do not yet have a feature freeze, and there was never a decision to add NO new component to MSL 3.2.1

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 16 May 2013 14:29 UTC**
Model: `Modelica.Media.Examples.RealGasAir.MoistAir1` redeclares medium to be:
```
redeclare package Medium = Modelica.Media.Air.RealGasMoistAir;
```
and `Modelica.Media.Air.RealGasMoistAir` contains:
```
package Modelica.Media.Air.RealGasMoistAir
  extends Modelica.Media.Interfaces.PartialRealCondensingGases(
    fluidConstants={Modelica.Media.Water.IAPWS95.Water95_Base.waterConstants,
        Modelica.Media.Air.RealGasAir.Air_Base.airConstants});
```

And `Modelica.Media.Water.IAPWS95.Water95_Base` is a partial package.

There are plenty of these patterns in Media, see also #1106 (not as obvious as this one). I have nothing against using stuff from partial packages directly via qualified names but the specification seems not to allow it.

Cheers,
Adrian Pop/

**Comment by modelica-trac-importer** [comment]
**Comment by wischhusen on 17 May 2013 11:18 UTC**
Replying to [comment:8 dietmarw]:

I had a look at the HTML issues. Here is the result for Modelica.Media:

In almost 10000 lines there are 12 lines with warnings for Modelica.Media. My hypothesis from the warning found

```
Warning: unescaped & or unknown entity "&QUOT"
```

is that the check routine does not verfiy upper case HTML commands. In fact the command &QUOT; used in Modelica.Media is right when a tool is not case sensitive. There are plenty of warnings like that for other parts of MSL. If I am right the solution for Modelica.Media would be to change 
```
&QUOT;
```
 into 
```
&quot;
```
Probably it is best to test it. If it resolves the problem it should be mentioned in the HTML guidelines that one has to use lower case commands. 

Thanks to Leo for adding capital letters in the description strings.

 

> Since some time we are struggling to improve the quality of the MSL. True the current MSL does not follow the guidelines 100%, that's why we said that all that is **added** to it follows this so that we only need to worry about updating existing code. It's much easier to fix new things at once than to go over it after some time. You say it is impossible by a human to follow the guidelines? They are not that much to read really and also quite easy to follow. If we can not manage even that tiny bit then I really don't want to think about what else we don't follow when adding new models.
> 
> In particular I meant that:
>  \* Description strings should start with a capital letter.
>  \* HTML errors in the documentation (Search for `Modelica.Media` in the [tidy.filtered](https://build.openmodelica.org/Documentation/tidy.filtered) file that is generated by OpenModelica on a daily basis

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 17 May 2013 11:33 UTC**
Entity names are case sensitive. `&Aring;` (&Aring;) is not the same as `&aring;` (&aring;).
See http://www.w3.org/TR/html4/charset.html#entities and other HTML specs.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 17 May 2013 13:53 UTC**
Replying to [comment:18 wischhusen]:


> Replying to [comment:8 dietmarw]:
> 
> I had a look at the HTML issues. Here is the result for Modelica.Media:
> 
> In almost 10000 lines there are 12 lines with warnings for Modelica.Media. 

I fixed the html-errors already yesterday.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 17 May 2013 15:10 UTC**
Replying to [comment:20 otter]:
> Replying to [comment:18 wischhusen]:
> In almost 10000 lines there are 12 lines with warnings for Modelica.Media. 

The HTML errors that are reported by [tidy.filtered](https://build.openmodelica.org/Documentation/tidy.filtered) are not just for the MSL but for all public libraries as listed on https://modelica.org/libraries. Hence the large number of warnings. Warnings for MSL should always be kept at 0.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 17 May 2013 17:46 UTC**
Stefan, Please pick up how the issue with partial types was changed for moist air. All Types are now defined in
```
 Modelica.Media.Interfaces.Types
```
Which is non-partial. If you use basic types and composite ones like FluidConstants from there, everything will be conformant to the specification. I will move this issue to #1106, and assign it to you, Stefan.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 17 May 2013 18:26 UTC**
There are a few more issues with the code included here.

 * This has been adapted from the ThermoFluid library by Jonas Eborn, Falko Jens Wagner and me. There should be a copyright notice that says that, copyright is not wth Lund University.
 * There are a number of functions and record definitions in Media.R134a.Common which are duplicates (or functional equivalents, because the versions in Modelica.Media evolved) of what is available in Modelica.Media.Common. This should absolutely not be done this way in the MSL, the functions should be removed and the functions/records in Modelica.Media.Common should be referenced. 

Sorry that I did not have time to review this earlier.

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 17 May 2013 19:16 UTC**
There is one more issue about naming. The new Air real gas model uses a Helmholtz EOS, which is a very good, high accuacy EOD, but is usually not referred to as "real gas" model. The most typical EOS for real gases are cubic ones, or the virial EOS. If we later on add other EOS to Modelica.Media, which could be used for air, are much faster than the Helmholtz equation and much more accurate at high pressure than ideal gases, the proper name would already be used up. I suggest something like "ReferenceAir", because it is the msot accrate EOS that I now of. 

I would strongly suggest to change the name before we actually make the inclusion.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 17 May 2013 20:22 UTC**
In 2f0021815c1742dd1ed6b6ae0c04b8dbef05be02 I fixed some non-ASCII issues. Please keep the MSL ASCII.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 18 May 2013 10:02 UTC**
Replying to [comment:24 hubertus]:
> There is one more issue about naming. The new Air real gas model uses a Helmholtz EOS, which is a very good, high accuacy EOD, but is usually not referred to as "real gas" model. The most typical EOS for real gases are cubic ones, or the virial EOS. If we later on add other EOS to Modelica.Media, which could be used for air, are much faster than the Helmholtz equation and much more accurate at high pressure than ideal gases, the proper name would already be used up. I suggest something like "ReferenceAir", because it is the msot accrate EOS that I now of. 
> 
> I would strongly suggest to change the name before we actually make the inclusion. 

Fixed in d1080b7f1fd11571f7251b287b37a2e4ba58a531 by changing the name to ReferenceAir.

Question: Should the RealGasMoistAir name then also be changed, say, to ReferenceMoistAir?

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 18 May 2013 18:39 UTC**
Yes, that would be the consistent solution.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 20 May 2013 09:14 UTC**
Replying to [comment:23 hubertus]:


> There are a few more issues with the code included here.
> 
> \* This has been adapted from the ThermoFluid library by Jonas Eborn, Falko Jens Wagner and me. There should be a copyright notice that says that, copyright is not wth Lund University.

Fixed in b9335ebd3d36efafe7686ac847e4c4950ed4b265 by providing the corrected copyright notice.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 20 May 2013 10:04 UTC**
Replying to [comment:23 hubertus]:

>  \* There are a number of functions and record definitions in Media.R134a.Common which are duplicates (or functional equivalents, because the versions in Modelica.Media evolved) of what is available in Modelica.Media.Common. This should absolutely not be done this way in the MSL, the functions should be removed and the functions/records in Modelica.Media.Common should be referenced. 

Fixed in c7583972333b650a6492667aff1d661cf7632778, f6cb1f06795ca774971dcbc00993507c590c930c, f8066f7c7c80a0f4eb2197fb423e109e6c66fde6:

Removed the definitions of NewtonDerivatives_ph, NewtonDerivatives_ps, HelmholtzDerivs, Helmholtz_ph in R134a and replaced references by the appropriate Media.Commons.XXX element.

The elements cv2Phase, helmholtzToBoundaryProps, and PhaseBoundaryProperties have the same names as in Media.Commons, but they have more elements and can therefore not be removed

**Comment by modelica-trac-importer** [comment]
**Comment by wischhusen on 20 May 2013 18:59 UTC**
Replying to [comment:27 hubertus]:
> Yes, that would be the consistent solution. 

Done with 920f38b942b1d71ddbc7295c57846de639faf3ca.

**Comment by modelica-trac-importer** [comment]
**Comment by leo&#46;gall on 21 May 2013 09:22 UTC**
I created a new ticket #1115 for the lower case description strings, because this matters for all classes of Modelica.Media (not only MoMoLib).

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 21 May 2013 09:56 UTC**
According to my understanding all reported issues are fixed and therefore this ticket is closed. If a new issue is reported (or some issue already reported overseen) it seems better to just make a new ticket.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 22 May 2013 13:06 UTC**
I just ran the HTML check again and some HTML references need to be updated/corrected:
```
[Checking file Modelica.Media.Air.ReferenceAir.html]:
Not found: Modelica.Media.Air.RealGasAir.Air_dT.html
[Checking file Modelica.Media.Air.ReferenceAir.html]:
Not found: Modelica.Media.Air.RealGasAir.Air_pT.html
[Checking file Modelica.Media.Air.ReferenceAir.html]:
Not found: Modelica.Media.Air.RealGasAir.Air_ph.html
```

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 22 May 2013 13:20 UTC**
Fixed in 7dcaa61ccd9f1d5ae0e751e0d0ce3d4a0064743e (7 occurrences of RealGasAir in Modelica replaced by ReferenceAir)
